napari/layers/base/base.py

```diff
@@ -1258,6 +1258,35 @@ def _display_bounding_box(self, dims_displayed: np.ndarray):
         """An axis aligned (self._ndisplay, 2) bounding box around the data"""
         return self._extent_data[:, dims_displayed].T
 
+    def click_plane_from_click_data(
+        self,
+        click_position: np.ndarray,
+        view_direction: np.ndarray,
+        dims_displayed: List,
+    ) -> Tuple[np.ndarray, np.ndarray]:
+        """Calculate a (point, normal) plane parallel to the canvas in data
+        coordinates, centered on the centre of rotation of the camera.
+
+        Parameters
+        ----------
+        click_position : np.ndarray
+            click position in world coordinates from mouse event.
+        view_direction : np.ndarray
+            view direction in world coordinates from mouse event.
+        dims_displayed : List
+            dimensions of the data array currently in view.
+
+        Returns
+        -------
+        click_plane : Tuple[np.ndarray, np.ndarray]
+            tuple of (plane_position, plane_normal) in data coordinates.
+        """
```

Comment:
This says that the click position is in "world coordinates". Could you elaborate what that means? If it's just any position in world space, why is it specifically "click position"?

Reply:
I've had trouble with this too - the key is that `event.position` from the mouse event is a position in nD world coordinates. That position is on a plane, centered on the `Camera.center`, parallel to the canvas. This method transforms this plane (defined by the click position and view direction) into data coordinates so you can use it to query things in the data coordinate system. Does that help?

Reply:
Yup, that helps! I propose to add some of this to the docstring, so it's clear what "click_position" represents: not just *any* position, but one that's on a plane passing through `Camera.center` and parallel to the canvas.